Externalize ha-assist-chat styling

balloob opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue or discussion
  in the additional information section.
-->
Move the margin out of `ha-assist-chat` so it can be easier embedded in other locations. Styling of assist dialog remains the same.

![CleanShot 2025-01-12 at 16 38 59@2x](https://github.com/user-attachments/assets/f4a5d282-a1f0-47f8-9a5c-428a0e203f0a)

Will help with styling for #23682

## Type of change
<!--
  What type of change does your PR introduce to the Home Assistant frontend?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR.
-->

```yaml

```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

<!--
  Thank you for contributing <3
-->

[docs-repository]: https://github.com/home-assistant/home-assistant.io

balloob committed:
Externalize ha-assist-chat styling

balloob commented on src/dialogs/voice-command-dialog/ha-voice-command-dialog.ts:
```diff
@@ -293,11 +293,14 @@ export class HaVoiceCommandDialog extends LitElement {
           display: flex;
           justify-content: center;
         }
+        ha-assist-chat {
```
@wendevlin I am not happy with this solution as we shouldn't mess with styling inside components. I couldn't find anything else in the brief time I had allocated for this. Got an idea? (Feel free to push to this PR)

wendevlin replied:
Mhh, but the title of your PR says exactly this that you want to change styling outside. We could work with a variable to set default values.

balloob replied:
Exactly. Styling a custom elmeent on the outside (margin) is fine. Adding padding to an element means that the element could break this accidentally in the future. However in this case, I am unable to get it working with the required flex box/height without triggering a scroll wheel.

bramkragten commented on src/components/ha-assist-chat.ts:
```diff
@@ -485,7 +485,7 @@ export class HaAssistChat extends LitElement {
       ha-textfield {
         display: block;
-        margin: 0 24px 16px;
+        padding: var(--ha-assist-chat-textfield-padding, 0 24px 16px);
```
This used to be margin, why was it changed to padding?

balloob replied:
Yep, we should never change the padding inside another custom element using CSS. I'm also not a big fan of allowing this padding to be overridden like this. The styling inside the element should just remain the same, it's only the border around it that is preferably gone and up to the host element to define.

wendevlin replied:
ok yeah I was a bit confused what this PR is about. You just want to set a margin around it :+1:

balloob commented:
I pushed a fix. The problem was that `ha-assist-chat` had a min-height, dictating how it was to be used. However, it was going to be used slightly different now. Removed the min-height and moved that to the host.

wendevlin commented on src/components/ha-assist-chat.ts:
```diff
@@ -498,7 +496,7 @@ export class HaAssistChat extends LitElement {
         right: 0px;
         left: 0px;
-        padding: 24px;
+        padding-bottom: 16px;
```
The x padding added a space betweren the chat bubbles and the scrollbar. Now it sticks to the scrollbar

balloob replied:
Ugh good point.

balloob replied:
Fixed.

balloob committed:
Don't mess with newlines

balloob committed:
Fix chat scrollbar spacing
